public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -62,6 +69,11 @@ async function fetchDashboard(
         // load home dash
         const dashDTO: DashboardDTO = await backendSrv.get('/api/dashboards/home');
 
+        // if above all is cancelled it will return an array
+        if (!dashDTO.meta) {
+          return null;
+        }
```

Comment:
had to add this, as this call get's cancelled as well by the cancellation of all in-flight requests. Thin backendSrv.get is not behaving a bit differently, after the switch to rxjx, it does not throw error with cancelled = true anymore as it did before, but now returns an empty array on cancellation. Think we need to change this so that cancellations return an error. But can revisit that in another PR that this :)

Reply:
That was a deliberate change we introduced when we introduced cancellation to the request function here https://github.com/grafana/grafana/pull/22066. I remember that we had a discussion (outside of Github) and we decided to not throw and just return an empty array[]. But if you feel differently about it now of course we should revert that change in a future PR.

Reply:
Right now cancelAllInFlightRequests cancels all requests. We could separate cancellation of `request` and `dataSourceRequest` so in this case the api call wouldn't be cancelled. Don't know if that is a better solution? What do you think?

Reply:
> What do you think @mckn @torkelo, should we merge this one and work on a better approach for 7.1? Also what should we do with the discussion about cancellation above [#24430 (comment)](https://github.com/grafana/grafana/pull/24430#discussion_r427744365)? Should I open a new issue for that?

Yes, I think it will bring a lot of value in its current form. Go go go :)